db/migrate/20210512113443_add_username_to_users.rb

```diff
@@ -0,0 +1,6 @@
+class AddUsernameToUsers < ActiveRecord::Migration[6.1]
+  def change
+    add_column :users, :username, :string, null: false
+    add_index :users, :username, unique: true
```

Comment:
I see, it's defined as unique here. But either way, I used to prefer login using email vs username XD 
But it's just a personal opinion. Please take it like a grain of salt